tests/unit/package_managers/test_general.py

```diff
@@ -132,3 +143,69 @@ def test_extract_git_info(url: str, nonstandard_info: Any) -> None:
     }
     info.update(nonstandard_info or {})
     assert general.extract_git_info(url) == info
+
+
+@pytest.mark.asyncio
+async def test_async_download_binary_file(tmp_path: Path) -> None:
+    url = "http://example.com/file.tar"
+    download_path = tmp_path / "file.tar"
+    chunk_size = 8192
+
+    # Create a mock response
+    response = MagicMock()
+
+    response.content.read = MockReadChunk().read_chunk
+
+    # Create a mock session
+    session = MagicMock()
+
+    async def mock_aenter() -> MagicMock:
+        return response
+
+    session.get().__aenter__.side_effect = mock_aenter
+
+    await _async_download_binary_file(session, url, download_path, chunk_size=chunk_size)
+
+    # Verify that the file was downloaded correctly
+    with open(download_path, "rb") as f:
+        assert f.read() == b"first_chunk-second_chunk-"
+
+    # # Verify that session.get was called with the correct arguments
+    assert session.get.called
+    assert session.get.call_args == mock.call(url, auth=None, raise_for_status=True)
+
+
+@pytest.mark.asyncio
+async def test_async_download_files(
+    tmp_path: Path, mock_async_download_binary_file: MagicMock
+) -> None:
+    files_to_download: Dict[str, Union[str, PathLike[str]]] = {
+        "file1": str(tmp_path / "path1"),
+        "file2": str(tmp_path / "path2"),
+        "file3": str(tmp_path / "path3"),
+    }
+
+    concurrency_limit = 2
+
+    # Mock aiohttp session and retry client
+    mock_session = MagicMock(spec=aiohttp.ClientSession)
+    mock_retry_client = MagicMock(spec=aiohttp_retry.RetryClient)
+
+    # Patch the relevant functions
+    with patch("aiohttp.ClientSession", return_value=mock_session), patch(
+        "aiohttp_retry.RetryClient", return_value=mock_retry_client
+    ), patch(
+        "cachi2.core.package_managers.general._async_download_binary_file",
+        new=mock_async_download_binary_file,
+    ) as mock_download_file:
```

Comment:
Is it possible to change this for the mock patch annotation, like we do in other tests ([e.g.](https://github.com/containerbuildsystem/cachi2/blob/main/tests/unit/package_managers/test_npm.py#L587-L588))?